### PR TITLE
Add color-hex-alpha rule

### LINKED ISSUE #4880: Add color-hex-alpha
> What is the problem you're trying to solve?

Some of the browsers not supported #rgba and #rrggbbaa kind of hex codes for colors.
refer https://caniuse.com/#feat=css-rrggbbaa

> What solution would you like to see?

So it would be great if we have an option in color-hex-length to prevent 4 & 8 digit hex colors.

### PR DESCRIPTION
**Pull request by ssivanatarajan**
> Which issue, if any, is this issue related to?

Have added a new rule as discussed in issue #4880. 

> Is there anything in the PR that needs further explanation?

No need

### CODE REVIEW

File: lib/rules/color-hex-alpha/__tests__/index.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	reject: [
+		{
+			code: 'a{color:#8888;}',
+			message: messages.unexpected('#8888'),
+			line: 1,
+			column: 9,
+		},
+	],
+});
+testRule({
```
Comment (jeddy3) [inline lib/rules/color-hex-alpha/__tests__/index.js:25]:
```suggestion
});

testRule({
```

File: lib/rules/color-hex-alpha/index.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+'use strict';
+
+const declarationValueIndex = require('../../utils/declarationValueIndex');
+const isStandardSyntaxDeclaration = require('../../utils/isStandardSyntaxDeclaration');
+const isStandardSyntaxValue = require('../../utils/isStandardSyntaxValue');
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+const valueParser = require('postcss-value-parser');
+const ruleName = 'color-hex-alpha';
```
Comment (jeddy3) [inline lib/rules/color-hex-alpha/index.js:11]:
```suggestion
const valueParser = require('postcss-value-parser');

const ruleName = 'color-hex-alpha';
```

File: lib/rules/color-hex-alpha/index.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+			actual: primary,
+			possible: ['always', 'never'],
+		});
+
+		if (!validOptions) return;
+
+		root.walkDecls((decl) => {
+			if (!isStandardSyntaxDeclaration(decl)) return;
+
+			const parsedValue = valueParser(getValue(decl));
```
Comment (jeddy3) [inline lib/rules/color-hex-alpha/index.js:30]:
```suggestion
			const parsedValue = valueParser(decl.value));
```

`decl.value` will suffice as we're not changing the AST.

File: lib/rules/color-hex-alpha/index.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+				}
+			});
+		});
+	};
+}
+
+function getValue(decl) {
+	return decl.raws.value ? decl.raws.value.raw : decl.value;
+}
+
```
Comment (jeddy3) [inline lib/rules/color-hex-alpha/index.js:66]:
```suggestion
```

File: lib/rules/color-hex-alpha/__tests__/index.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+	],
+	reject: [
+		{
+			code: 'a{color:#888;}',
+			message: messages.expected('#888'),
+			line: 1,
+			column: 9,
+		},
+	],
+});
```
Comment (jeddy3) [inline lib/rules/color-hex-alpha/__tests__/index.js:44]: Let's get an additional `testRule` block with `syntax: "scss" and an interpolation accept test case, e.g.

```css
a { color: #{ff}; }
```

```js
testRule({
	ruleName,
	config: ["never"],
	syntax: 'scss',

	accept: [
		{
			code: 'a { color: #{ff}; }',
			description: 'scss interpolation',
		},
	],
});

**Comment by ssivanatarajan** [comment]
@jeddy3 Made the changes Please review it.

**Comment by ssivanatarajan** [comment]
@jeddy3 can u please review this PR

**Comment by ssivanatarajan** [comment]
@jeddy3  any update on this ?

**Comment by ssivanatarajan** [comment]
> @ssivanatarajan Thanks for the pull request. It's looking good.
> 
> I've requested some changes.
> 
> Can you also add the rule [to the list in the docs](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/rules/list.md), please?

added

**Comment by ssivanatarajan** [comment]
@jeddy3 is anything that needs to be changed /reviewed?

**Review by jeddy3 — CHANGES_REQUESTED** [review]
Thanks for making the changes.

I've looked through the PR again and requested some more changes.

**Comment by jeddy3** [comment]
Finished in #5316

**Pull request closed without merging**